app/models/assignment.rb

```diff
@@ -8,15 +8,31 @@ class Assignment < ApplicationRecord
   }, if: :lti_integrated?
   belongs_to :group
   has_many :projects, class_name: "Project", dependent: :nullify
+  before_destroy :cleanup_notification
 
   after_commit :send_new_assignment_mail, on: :create
   after_commit :set_deadline_job
   after_commit :send_update_mail, on: :update
+  after_create_commit :notify_recipient
 
   enum grading_scale: { no_scale: 0, letter: 1, percent: 2, custom: 3 }
   default_scope { order(deadline: :asc) }
   has_many :grades, dependent: :destroy
 
+  has_noticed_notifications model_name: "NoticedNotification"
+
+  def notify_recipient
+    @assignment = Assignment.find(id)
+    group.group_members.each do |group_member|
+      # NewAssignmentNotification.with(user: group_member.user, assingment: @assingment).deliver(group_member.user)
+      NewAssignmentNotification.with(assingment: @assignment).deliver(group_member.user)
```

Comment:
```suggestion
      NewAssignmentNotification.with(assignment: @assignment).deliver_later(group_member.user)
```

- `assingment` spelling is not correct!
- use `deliver_later` instead